app/src/test/java/org/simple/clinic/editpatient/PatientEditScreenCreatedWithDataTest.kt

```diff
@@ -0,0 +1,165 @@
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.never
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.whenever
+import io.reactivex.Observable
+import io.reactivex.subjects.PublishSubject
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.simple.clinic.patient.Age
+import org.simple.clinic.patient.PatientMocker
+import org.simple.clinic.patient.PatientRepository
+import org.simple.clinic.registration.phone.PhoneNumberValidator
+import org.simple.clinic.util.Just
+import org.simple.clinic.util.RxErrorsRule
+import org.simple.clinic.util.TestUserClock
+import org.simple.clinic.util.TestUtcClock
+import org.simple.clinic.util.toOptional
+import org.simple.clinic.widgets.UiEvent
+import org.simple.clinic.widgets.ageanddateofbirth.UserInputDateValidator
+import org.threeten.bp.Instant
+import org.threeten.bp.LocalDate
+import org.threeten.bp.format.DateTimeFormatter
+import java.util.Locale
+
+@RunWith(JUnitParamsRunner::class)
+class PatientEditScreenCreatedWithDataTest {
+
+  @get:Rule
+  val rxErrorsRule = RxErrorsRule()
+
+  private val uiEvents = PublishSubject.create<UiEvent>()
+  val utcClock: TestUtcClock = TestUtcClock()
+
+  private lateinit var screen: PatientEditScreen
+  private lateinit var patientRepository: PatientRepository
+  private lateinit var numberValidator: PhoneNumberValidator
+  private lateinit var controller: PatientEditScreenController
+
+  private lateinit var errorConsumer: (Throwable) -> Unit
+  private lateinit var dobValidator: UserInputDateValidator
+  private val dateOfBirthFormat = DateTimeFormatter.ofPattern("dd/MM/yyyy", Locale.ENGLISH)
+
+  @Before
+  fun setUp() {
+    screen = mock()
+    patientRepository = mock()
+    numberValidator = mock()
+    dobValidator = mock()
+
+    whenever(dobValidator.dateInUserTimeZone()).thenReturn(LocalDate.now(utcClock))
+
+    controller = PatientEditScreenController(
+        patientRepository,
+        numberValidator,
+        utcClock,
+        TestUserClock(),
+        dobValidator,
+        dateOfBirthFormat)
+
+    errorConsumer = { throw it }
+
+    uiEvents
+        .compose(controller)
+        .subscribe({ uiChange -> uiChange(screen) }, { e -> errorConsumer(e) })
+  }
+
+  @Test
+  @Parameters(method = "params for prefilling fields on screen created")
+  fun `when screen is created then the existing patient data must be prefilled`(data: TestData) {
+    val patient = data.patient
+    val address = data.address
+    val shouldSetColonyOrVillage = data.shouldSetColonyOrVillage
+    val shouldSetPhoneNumber = data.shouldSetPhoneNumber
+    val shouldSetAge = data.shouldSetAge
+    val shouldSetDateOfBirth = data.shouldSetDateOfBirth
+    val phoneNumber = data.phoneNumber
+
+    whenever(patientRepository.patient(patient.uuid)).thenReturn(Observable.just(Just(patient)))
```

Comment:
These lines are no longer necessary, right?

Reply:
To get rid of this, we have to refactor the `savePatientDetails` function. Because it fetches patient information from the repository during the `PatientEditScreenCreated`. I can make this change as part of the next PR.

Reply:
👍